accepted/2020/workloads/workloads.md

```diff
@@ -0,0 +1,92 @@
+
+The target framework will be used as the primary currency in project files that establish a link to a workload. As part of a related effort, [target frameworks are being updated in .NET 5](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md) to include operating system information. This change will enable a direct link to OS-specific workloads. For example, a project that targets `net5-android` will need the .NET 5 Android workload (Xamarin.Android). A project that multi-targets `net5-android` and `net5-ios` will also need the .NET 5 iOS workload (Xamarin.iOS).
+
+The set of TFM to workload mappings will be finite, and defined by the workloads. The SDK will carry a copy of this mapping, but will defer to information carried as part of installed workloads when they present newer information.
+
+### Acquisition
+
+Workloads will be acquired in two primary ways: via a supported installation tool or the .NET CLI.
+
+Note: The term "supported installation tool" is intended to mean an installation system that exposes components and manages their dependencies. For example, Visual Studio Installer, Docker, and APT do that, while MSI and PKG don't.
+
+Install tools must follow these rules for workloads it installs:
+
+* The workload manifest is included as a required dependency for all variants of the workload (some packs might be optional).
+* All referenced packs are included, matching the version specified in the workload manifest.
+
+Users should be able to install workloads via an install tool without any sense that there is a compositional model at play.
+
+The .NET CLI will expose a more extensive set of commands for interacting with workloads. The CLI may allow installing more than just workloads, such that the noun it exposes is not `workload` but something more general (TBD; `bundle` is used as the stand-in for a better term in the following examples). In this parlance, a workload is one kind of bundle.
```

Comment:
I really dislike introducing a new word. I've preferred workload to optional component all along, but let's keep the number of words to a minimum. (replace bundle with workload here, imo)

Reply:
I tend to agree - it seems like the CLI experience would be more intuitive if we used the same words.

> In this parlance, a workload is one kind of bundle.

This seems to imply there are other types of bundles, is this just future proofing or is there something else in mind?

Reply:
This is a new term. We have three levels:

- workloads (like Xamarin Android)
- The curated bundles of packs it exposes, one of which will be default, and the others optional.
- The packs

We could call the middle thing "components" but the middle thing doesn't really align with my definition of "component". A component usually has a well-defined interface and interaction model. The bundles lack that.

I got feedback from some else that sent me down the bundle path.